src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/VsTelemetryService.cs

```diff
@@ -39,5 +161,88 @@ public TelemetryEventCorrelation Report(string eventName, string description, Ex
                 exceptionObject: exception,
                 gatherEventDetails: callback);
         }
+
+        /// <summary>
+        /// Hashes personally identifiable information for telemetry consumption.
+        /// </summary>
+        /// <param name="value">Value to hashed.</param>
+        /// <returns>Hashed value.</returns>
+        public string HashValue(string value)
+        {
+            using (var crypto = SHA256.Create())
```

Comment:
According to [this](https://crypto.stackexchange.com/questions/3859/fips-140-2-compliant-algorithms) and links inside, SHA256 is FIPS compliant.